### PR TITLE
"Support" for "dayNightSpeed" and "speed" command

### PR DESCRIPTION
**Pull request by Jannify**

### CODE REVIEW

**Comment by tornac1234** [comment]
where is my co-authored tag :( xD

**Comment by Jannify** [comment]
> where is my co-authored tag :( xD

Didn't know this was possible with such little effort, here it is:

**Comment by Jannify** [comment]
🙃

File: NitroxPatcher/Patches/Dynamic/DayNightCycle_OnConsoleCommand_daynightspeed_Patch.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    // The command is skipped because simulating speed reliable on the server is out of scope
+    public static bool Prefix()
+    {
+#if DEBUG
+        ErrorMessage.AddMessage("This command is not fully supported. Expect breaking your instance/world.");
+        return true;
+#else
+        ErrorMessage.AddMessage(Language.main.Get("Nitrox_CommandNotAvailable"));
+        return false;
+#endif
```
Comment (dartasen) [inline NitroxPatcher/Patches/Dynamic/DayNightCycle_OnConsoleCommand_daynightspeed_Patch.cs:20]: Do we need to enable them in DEBUG ?

**Review by dartasen — CHANGES_REQUESTED** [review]
Maybe we should start looking at extending our command system so we can wrap and filter every vanilla commands by patching the NotificationCenter once, instead of multiplying patches (time, sub, speed, ...)

File: NitroxPatcher/Patches/Dynamic/DayNightCycle_OnConsoleCommand_daynightspeed_Patch.cs (hunk 1 above)
Comment (Jannify) [reply NitroxPatcher/Patches/Dynamic/DayNightCycle_OnConsoleCommand_daynightspeed_Patch.cs:20]: We don't need it right now. Will change it, if somebody needs the command we devs can disable it manually.

**Comment by Jannify** [comment]
> Maybe we should start looking at extending our command system so we can wrap and filter every vanilla commands by patching the NotificationCenter once, instead of multiplying patches (time, sub, speed, ...)

Yeah, we should probably open a issue for this.

File: Nitrox.Assets.Subnautica/LanguageFiles/English.json [hunk 2]
```diff
@@ -40,5 +40,6 @@
 "Nitrox_WaitingPassword": "Waiting for server password input...",
 "Nitrox_WaitingUserInput": "Waiting for user input...",
 "Nitrox_LaunchGame": "Launching game...",
-"Nitrox_SilencedChatNotif": "Chat is now muted"
+"Nitrox_SilencedChatNotif": "Chat is now muted",
+"Nitrox_CommandNotAvailable": "This command is not available in Nitrox."
```
Comment (dartasen) [inline Nitrox.Assets.Subnautica/LanguageFiles/English.json:44]:
```suggestion
"Nitrox_CommandNotAvailable": "This command is not available in Nitrox"
```

File: Nitrox.Assets.Subnautica/LanguageFiles/German.json [hunk 3]
```diff
@@ -40,5 +40,6 @@
 "Nitrox_WaitingPassword": "Warte auf Eingabe des Server Passworts...",
 "Nitrox_WaitingUserInput": "Warte auf Nutzereingaben...",
 "Nitrox_LaunchGame": "Starte das Spiel...",
-"Nitrox_SilencedChatNotif": "Chat ist stummgeschaltet"
+"Nitrox_SilencedChatNotif": "Chat ist stummgeschaltet",
+"Nitrox_CommandNotAvailable": "Dieser Befehl ist in Nitrox nicht verfügbar."
```
Comment (dartasen) [inline Nitrox.Assets.Subnautica/LanguageFiles/German.json:44]:
```suggestion
"Nitrox_CommandNotAvailable": "Dieser Befehl ist in Nitrox nicht verfügbar"
```

**Review by dartasen — CHANGES_REQUESTED** [review]
Is it possible to add this ErrorMessage to sub command also ?

**Here's some translations I can give you :**
French : "Cette commande n'est pas disponible avec Nitrox"
Spanish: "Este comando no está disponible con Nitrox"
Portuguese: "Este comando não está disponível com Nitrox"
Italian: "Questo comando non è disponibile con Nitrox"

Maybe we can fill up others with Deepl ?

**Comment by Jannify** [comment]
As we are constantly adding now lines to the files I wouldn't set the priority to translating with tools. Imo we can wait for the native speaker to fill in the gaps. Especially if we want to switch to a online service in the future.

**Review by dartasen — APPROVED** [review]

**Review by Coding-Hen — APPROVED** [review]

**Comment by tornac1234** [comment]
Tornac approved these changes

**Review by tornac1234 — APPROVED** [review]

**Pull request merged**